homeassistant/components/discord/notify.py

```diff
@@ -39,30 +39,53 @@ def __init__(self, hass, token):
 
         discord.VoiceClient.warn_nacl = False
         discord_bot = discord.Client(loop=self.hass.loop)
+        images = None
 
         if ATTR_TARGET not in kwargs:
             _LOGGER.error("No target specified")
             return None
 
+        if ATTR_DATA in kwargs:
+            data = kwargs.get(ATTR_DATA)
+
+            if ATTR_IMAGES in data:
+                import os.path
+                images = list()
+
+                for image in data.get(ATTR_IMAGES):
+                    if os.path.isfile(image):
```

Comment:
This does I/O and we're in a coroutine. That combo is not allowed. Use `hass.async_add_executor_job` to schedule the I/O on the executor thread pool.